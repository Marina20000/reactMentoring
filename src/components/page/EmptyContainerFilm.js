import React, {PureComponent} from 'react';
import style from './style.css';

export default class EmptyContainerFilm extends PureComponent{

    render() {
        return <div className={style.emptyResult}>Find your movie</div>;
    }
}