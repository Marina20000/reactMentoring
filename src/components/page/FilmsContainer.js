import React from 'react';
import Flexbox from 'flexbox-react';
import MovieItem from './../movieItems/MovieItem';

export default class FilmsContainer extends React.Component {
  constructor(props) {
    super(props);
  }


  render() {
    let images = [];
    let movies = !!this.props.movies?this.props.movies:[];
    movies.forEach(index => {
      images.push(<MovieItem params={index} key={index.id} />)
    })
    return (<Flexbox flexDirection="column" minHeight="100vh" width = "100%">
      <Flexbox element="header" height="60px">
      </Flexbox>
      <Flexbox flexGrow={1} flexWrap={'wrap'} justifyContent={'center'}>
        {images}
      </Flexbox>
      <Flexbox element="footer" height="60px">
      </Flexbox>
    </Flexbox>);
  };
}
