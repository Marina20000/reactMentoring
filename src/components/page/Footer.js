import React from 'react';
import Inscription from './../captions/Inscription';
import style from './style.css';

export default class Footer extends React.Component {

    render() {
        return <div style={{ height: 100, backgroundColor: 'black', bottom: 0 }}>
            <Inscription className={ style.netflix } inscription='netflixroulette'/>
       </div>;
    }

}