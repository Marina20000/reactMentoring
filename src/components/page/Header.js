import React from 'react';
import SearchPanel from './../textboxes/SearchPanel';
import Inscription from './../captions/Inscription';
import CreateButton from './../buttons/CreateButton';
import style from './style.css';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './../../actions/action';
import SearchResultPanel from './SearchResultPanel';


export class Header extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            title: true,
            genre: false,
            value: '',
            releaseDate: true,
            raiting: false
        };
    }

    toggleButton = (filter) => {
        this.setState({ title: !this.state.title, genre: !this.state.genre });
    };

    searchFilms = () => {
        let filter = this.state.title ? 'title':'genre';
        let sorting =this.state.releaseDate ? 'release_date':'raiting';
        this.props.actions.getData(this.request(filter,sorting));
    }

    request = (filter,sorting) => {
        return 'limit=20&search='+this.state.value+'&searchBy='+filter+'&sortBy='+sorting+'&sortOrder=desc';
    }

    handleChange = e => {
        this.setState({ value: e.target.value });
    }

    toggleSortings = () => {
        this.setState({ releaseDate: !this.state.releaseDate, raiting: !this.state.raiting });
    };

    render() {
        return (
            <div style={{ height: 500, backgroundImage: `url('\./image.jpg')`, backgroundSize: '100%' }}>
                <div className={style.container}>
                    <Inscription className={style.netflix} inscription='netflixroulette' />
                    <Inscription className={style.findYourMovie} inscription='FIND YOUR MOVIE' />
                    <SearchPanel onChange={this.handleChange} value={this.state.value}/>
                    <div style={{ width: '100%' }}>
                        <Inscription className={style.searchBy} inscription='SEARCH BY' />
                        <CreateButton onClick={() => { this.toggleButton('title') }} className={this.state.title ? style.selected : style.noneSelected} inscription='TITLE' />
                        <CreateButton onClick={() => { this.toggleButton('genre') }} className={this.state.genre ? style.selected : style.noneSelected} inscription='GENRE' />
                        <CreateButton onClick={() => { this.searchFilms() }} className={style.searchButton} inscription='SEARCH' />
                    </div>
                </div>
                <SearchResultPanel quantity={ !!this.props.movies ? this.props.movies.length : 0 } showPage2 = { this.props.showPage2 } toggleSortings={this.toggleSortings}
                raiting = { this.state.raiting } releaseDate = { this.state.releaseDate }/>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        movies: state.app.getFilms.data,
    }
  }
  
  function mapDispatchToProps(dispatch) {
    return {
      actions: bindActionCreators(actions, dispatch)
    }
  }
  
  export default connect(mapStateToProps, mapDispatchToProps)(Header);