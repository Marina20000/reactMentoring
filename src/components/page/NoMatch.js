import React from 'react';
import style from './style.css';

export default class NoMatch extends React.Component{
render() {
     return <div className={style.emptyResult}>
        Not found
    </div>;
}
}