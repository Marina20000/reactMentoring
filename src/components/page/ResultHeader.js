import React from 'react';
import MovieItem from './../movieItems/MovieImage';
import ContainerDescription from './../movieItems/ContainerDescription';
import Inscription from './../captions/Inscription';
import CreateButton from './../buttons/CreateButton';
import style from './style.css';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './../../actions/action';


export class ResultHeader extends React.Component {
    constructor(props) {
        super(props);
    }

    returnButtonClick = () => {
        this.props.actions.changePage();
    }

    render() {
        return (
            <div style={{ backgroundImage: `url('\./image.jpg')`, backgroundSize: '100%' }}>
                <div className={style.container}>
                    <Inscription className={style.netflix} inscription='netflixroulette' />
                    <CreateButton onClick={() => { this.returnButtonClick() }} className={style.returnButton} inscription='SEARCH' />
                    <img src={!!this.props.film ? this.props.film.poster_path : ''} className={style.movieImage} />
                    <ContainerDescription film={this.props.film} />

                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    console.log('state', state)
    return { film: state.clickFilm.film }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ResultHeader);