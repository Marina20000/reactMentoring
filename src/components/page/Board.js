import React from 'react';
import FilmsContainer from './FilmsContainer';
import SearchResultPanel from './SearchResultPanel';
import Inscription from './../captions/Inscription';
import style from './style.css';

export default class Board extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
        let searchResult = this.props.searchResult;
         return <div >
            { searchResult && <Inscription className={ style.noFilmsFound } inscription = 'No films found'/> }
            <FilmsContainer  movies = { this.props.movies }/>
        </div>;
    }
}