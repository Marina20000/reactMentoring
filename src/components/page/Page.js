import React from 'react';
import Header from './Header';
import Footer from './Footer';
import Board from './Board';
import ResultHeader from './ResultHeader';
import ErrorBoundary from './ErrorBoundary';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './../../actions/action';
import configureStore from 'redux-mock-store';

export class Page extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            movies: [],
        };
    }

    componentDidMount() {
        this.props.actions.getData();
    }

    render() {
        let searchResult = this.props.data == 0;
        let showPage2 = this.props.changePage;
        return (
            <div style={{ height: 1000 }}>
                <ErrorBoundary>
                    {showPage2 == false ? <Header movies={this.props.data} /> : <ResultHeader film={this.props.film} />}
                </ErrorBoundary>
                <ErrorBoundary>
                    <Board searchResult={searchResult} showPage2={showPage2} movies={this.props.data} />
                </ErrorBoundary>
                <Footer />
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        data: state.getFilms.data,
        film: state.clickFilm.film,
        changePage: state.changePage.showPage2
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Page);