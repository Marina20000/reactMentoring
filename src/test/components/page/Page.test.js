import React from "react";
import renderer from 'react-test-renderer';
import { Page } from './../../../components/page/Page';


const prettyFormat = require('pretty-format');
describe('Page', () => {
    let getData = jest.fn();
    let changePage =jest.fn();
    let actions = {getData: getData, changePage:changePage};
    let _props = {data:[],film:{},changePage:false, actions:actions};
    it("testing with snapshot", ()=>{
        global.fetch = jest.fn().mockImplementation(() => Promise.resolve({}));
        const page = shallow(<Page {..._props} />);       
        expect(page).toMatchSnapshot();
    })
})
