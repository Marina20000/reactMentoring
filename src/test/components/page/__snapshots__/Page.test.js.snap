// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page testing with snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page
    actions={
      Object {
        "changePage": [MockFunction],
        "getData": [MockFunction] {
          "calls": Array [
            Array [],
          ],
        },
      }
    }
    changePage={false}
    data={Array []}
    film={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ErrorBoundary>
          <Connect(Header)
            movies={Array []}
          />
        </ErrorBoundary>,
        <ErrorBoundary>
          <Board
            movies={Array []}
            searchResult={true}
            showPage2={false}
          />
        </ErrorBoundary>,
        <Footer />,
      ],
      "style": Object {
        "height": 1000,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(Header)
            movies={Array []}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movies": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Board
            movies={Array []}
            searchResult={true}
            showPage2={false}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movies": Array [],
            "searchResult": true,
            "showPage2": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ErrorBoundary>
            <Connect(Header)
              movies={Array []}
            />
          </ErrorBoundary>,
          <ErrorBoundary>
            <Board
              movies={Array []}
              searchResult={true}
              showPage2={false}
            />
          </ErrorBoundary>,
          <Footer />,
        ],
        "style": Object {
          "height": 1000,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Header)
              movies={Array []}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "movies": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Board
              movies={Array []}
              searchResult={true}
              showPage2={false}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "movies": Array [],
              "searchResult": true,
              "showPage2": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
